torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -189,7 +189,7 @@ class TORCH_API RpcAgent {
   virtual void sync() = 0;
 
   // start accepting requests
-  virtual void start() = 0;
+  virtual void start();
```

Comment:
Let's clearly state what is the contact for derived classes. In the current implementation, it looks like there are at least two requirements:

1. derived class must call start from `RpcAgent`
2. derived class must set `rpcAgentRuninng_` before calling `RpcAgent::start()`

We might be able to remove the 2nd if we consolidate the `rpcAgentRunning_` into `RpcAgent::start()`

Reply:
I think a non-virtual `start` and a pure virtual `startImpl` that subclasses need to override seems like a cleaner contract.